app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -1163,6 +1163,10 @@ BrowserWorld::StartFrame() {
     if (relayoutWidgets) {
       UpdateVisibleWidgets();
     }
+    if (m.device->IsPassthroughEnabled() && m.device->usesPassthroughCompositorLayer() && !m.layerPassthrough) {
```

Comment:
Just a suggestion; I find the long conditional clause a bit confusing. Wouldn't be clearer to move it to a keep just m.device->IsPassthroughEnabled(), move the rest to a createPassthoughLayerIfNeeded() method; possibly m.device->usesPassthroughCompositorLayer() would become an assert in that case (is it possible that passthough is enabled but we are not using the compositor layer ?